docs: Document `--debug-verbose=datapath` in debugging datapath section (#16022)
cilium-team/cilium: Documentation/contributing/development/dev_setup.rst

from the eBPF based datapath. Debugging messages are sent if either the
``cilium-agent`` itself or the respective endpoint is in debug mode. The debug
mode of the agent can be enabled by starting ``cilium-agent`` with the option
``--debug`` and ``--debug-verbose=datapath`` enabled or by running 
``cilium config debug=true`` for an already running agent. The 
``--debug-verbose=datapath`` option is necessary only if you want to trace the 

> `printk` messages in eBPF datapath. 

Reviewer: Some elements are not clear from this modification.
- What is `--debug-verbose=datapath` needed for? I.e. what does it do that `--debug` does not?
- Does `cilium config debug=true` also set `--debug-verbose=datapath`, or does it need to be set separately?

.. note::

>    The `printk` is available in `bpf/lib/dbg.h` file.

Reviewer: ```suggestion
   The `printk()` helper is available from the `bpf/lib/dbg.h` header file.
```

Reviewer: Please mark the conversations as “resolved” when they are. Makes it easier to see what's left to address :).

.. note::

>    You can use `bpftool prog trace` to show the `printk` messages.

Reviewer: ```suggestion
   You can use `bpftool prog tracelog` to show the `printk` messages.
```

You could group the two notes in a single “`printk`-usage” note, I think.

>  Debugging of an individual endpoint can be enabled by running

Reviewer: ```suggestion
Debugging of an individual endpoint can be enabled by running
```

Datapath code
^^^^^^^^^^^^^

The image below shows the options that could be used as startup options by
``cilium-agent`` (see upper blue box) or could be changed at runtime by running
``cilium config <option(s)>`` for an already running agent (see lower blue box).
For instance, if you want to enable the debug mode (including the load
balancer), you may start ``cilium-agent`` with ``--debug-verbose=datapath`` or
run ``cilium config debug=true debugLB=true`` for an already running agent. For
each option, there is a helper function which can be used by the developer to
print debugging information in the tool ``cilium monitor``: ``cilium_dbg()`` for

> ``DEBUG`` and ``cilium_dbg_lb()`` for ``DEBUG_LB``. 

Reviewer: Nice work! If I may, I think the beginning of this paragraph is a bit abrupt. Let's keep in mind that people coming to read that section are looking for a way to enable debug logs, because they likely have issues in their datapath: The first thing we give them is a big diagram with plenty of possibilities.

Let's try to bring it in a more gentle way, maybe something along:
```
The tool ``cilium monitor`` can also be used to retrieve debugging information
from the eBPF based datapath. It relies on ``cilium_dbg()`` and a few other related
function in the code. These functions are inactive by default and must be enabled
by passing the relevant options to the ``cilium-agent`` or by enabling it manually
from a running agent. The image below shows the details of ...
```
[Then give some info on the commands, as you do already.]

Another possibility could even be to provide the commands that enable the highest level of debugging first, and to explain things after that. Thinking of:

```
The tool ``cilium monitor`` can also be used to retrieve debugging information
from the eBPF based datapath. To enable all log messages:
- Start the ``cilium-agent`` with `--debug-verbose=datapath``, or
- Run ``cilium config debug=true debugLB=true`` from an already running agent.

These options enable logging functions in the datapath: `cilium_dbg()`, etc. ...
The image below shows...
```

What do you think?

Author: Thank you for your suggestions! Absolutely!

> .. image:: _static/cilium-debug-datapath-options.svg 

Reviewer: Nit: Any chance we could have lighter backgrounds (or white foreground) to improve readability for the dark grey and purple boxes?

Author: Let me know if it is as you expected.

  :align: center
  :alt: Cilium debug datapath options

.. note::

>    You can also use ``printk()`` to debug the datapath outside the ``cilium

Reviewer: ```suggestion
   You can also use ``printk()`` to debug the datapath outside of the ``cilium
```

   monitor``.  In this case, ``bpftool prog tracelog`` can be used to retrieve

>    debugging information from the eBPF based datapath.  However, it is recommended
>    to use ``cilium_dbg()``. Both ``cilium_dbg()`` and ``printk()`` helpers are

Reviewer: We would likely recommend `cilium_dbg()` over `prink()` because it is better integrated with the rest of the debug messages available from `cilium monitor`. But I don't think we should discourage people from using `prink()` if they want: When it comes to debugging, you never have too many tools. I think the `outside [of] the cilium monitor` sums up the idea already, so I'd just remove that sentence about the recommendation.

```suggestion
   debugging information from the eBPF based datapath.
   Both ``cilium_dbg()`` and ``printk()`` helpers are
```
